### PR TITLE
solution

### PR DESCRIPTION
**Pull request by vitvadym**
Had some troubles with pushing 😃

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const stateVersions = [];
+
+  for (const action of actions) {
+    const { type, keysToRemove, extraData } = action;
```
Comment (dinaminahi) [inline src/transformStateWithClones.js:14]: 👍

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
```
Comment (dinaminahi) [inline src/transformStateWithClones.js:10]: Let't make it const and then you should change the way you are clearing it

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -7,7 +7,38 @@
...
+        newState = {
+          ...newState,
+          ...extraData,
+        };
+        break;
+
+      case 'removeProperties':
+        for (const keyToRemove of keysToRemove) {
+          delete newState[keyToRemove];
+        }
```
Comment (dinaminahi) [inline src/transformStateWithClones.js:31]: Consider of creating separate function for this to make `switch-case` smaller and more readable

**Review by dinaminahi — CHANGES_REQUESTED** [review]
Well done 👍 
But few more things to improve it

**Review by naziamoff — APPROVED** [review]
🔥

**Pull request closed without merging**